Food_Tracker_App/app/src/main/java/com/example/foodtracker/model/recipe/Recipe.java

```diff
@@ -48,24 +49,45 @@ public String getCollectionName() {
     @Override
     public Map<String, Object> getData() {
         Map<String, Object> data = new HashMap<>();
-        data.put(Recipe.FieldNames.TITLE, this.getTitle());
-        data.put(Recipe.FieldNames.IMAGE, this.getImage());
-        data.put(Recipe.FieldNames.PREPTIME, this.getPrepTime());
-        data.put(Recipe.FieldNames.CATEGORY, this.getCategory());
-        data.put(Recipe.FieldNames.SERVINGS, this.getServings());
-        data.put(Recipe.FieldNames.COMMENT, this.getComment());
-        data.put(FieldNames.INGREDIENTS, this.getIngredients());
+        data.put(Recipe.FieldName.TITLE.getName(), this.getTitle());
+        data.put(Recipe.FieldName.IMAGE.getName(), this.getImage());
+        data.put(Recipe.FieldName.PREPARATION_TIME.getName(), this.getPrepTime());
+        data.put(Recipe.FieldName.CATEGORY.getName(), this.getCategory());
+        data.put(Recipe.FieldName.SERVINGS.getName(), this.getServings());
+        data.put(Recipe.FieldName.COMMENT.getName(), this.getComment());
+        data.put(FieldName.INGREDIENTS.getName(), this.getIngredients());
         return data;
     }
 
-    public static class FieldNames {
-        public static String TITLE = "title";
-        public static String IMAGE = "image";
-        public static String PREPTIME = "prepTime";
-        public static String CATEGORY = "category";
-        public static String SERVINGS = "servings";
-        public static String COMMENT = "comment";
-        public static String INGREDIENTS = "ingredients";
+    /**
+     * Represents the field names in the recipes class
+     */
+    public enum FieldName implements DocumentableFieldName {
+        TITLE("title", true),
+        IMAGE("image", false),
+        PREPARATION_TIME("prepTime", true),
```

Comment:
Need solution to make this work

Reply:
for now I will just not capitalize anything, I'll try to think of a solution that works to display the text in a prettier way